### PR TITLE
Add tf.data.Dataset type check in orca tf keras fit

### PR DESCRIPTION
**Pull request by cyita**

### CODE REVIEW

File: pyzoo/zoo/orca/learn/tf/estimator.py [hunk 1]
```diff
@@ -464,6 +464,17 @@ def fit(self, data,
             assert labels_cols is not None, \
                 "label columns is None; it should not be None in training"
 
+        if isinstance(data, tf.data.Dataset):
+            assert isinstance(data.element_spec, tuple), \
+                "If data is tf.data.Dataset, each element should be " \
+                "[feature tensor tuple, label tensor tuple]"
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/tf/estimator.py:470]: why do you mean by "[feature tensor tuple, label tensor tuple]"?

Comment (jenniew) [reply pyzoo/zoo/orca/learn/tf/estimator.py:470]: each element should be [feature tensor tuple, label tensor tuple] or (feature tensor tuple, label tensor tuple)

**Comment by cyita** [comment]
> what if there are multiple inputs (features) for the model?

changed to (feature tensor tuple, label tensor tuple)

File: pyzoo/zoo/orca/learn/tf/estimator.py [hunk 2]
```diff
@@ -467,13 +467,13 @@ def fit(self, data,
         if isinstance(data, tf.data.Dataset):
             assert isinstance(data.element_spec, tuple), \
                 "If data is tf.data.Dataset, each element should be " \
-                "[feature tensor tuple, label tensor tuple]"
+                "(feature tensor tuple, label tensor tuple)"
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/tf/estimator.py:470]: each element should be (feature tensors, label tensor), where each feature/label tensors can be either a single tensor or a tuple of tensors

**Comment by cyita** [comment]
Jenkins passed http://10.239.47.210:18888/job/ZOO-PR-Validation/4133/

**Pull request merged**